src/app/delivery-history/delivery-history.component.ts

```diff
@@ -65,15 +66,42 @@ export class DeliveryHistoryComponent implements OnInit {
 
     this.helperInfo = context.for(helperHistoryInfo, this.helperStorage).gridSettings({
       showFilter: true,
+      allowSelection: true,
 
       numOfColumnsInGrid: (this.settings.isSytemForMlt() ? 10 : 7),
       gridButtons: [{
-        name: this.settings.lang.exportToExcel,
-        visible: () => this.context.isAllowed(Roles.admin),
-        click: async () => {
-          await saveToExcel(this.settings, this.context.for(helperHistoryInfo, this.helperStorage), this.helperInfo, this.settings.lang.volunteers, this.busy, (d: helperHistoryInfo, c) => c == d.courier);
+          name: this.settings.lang.exportToExcel,
+          visible: () => this.context.isAllowed(Roles.admin),
+          click: async () => {
+            await saveToExcel(this.settings, this.context.for(helperHistoryInfo, this.helperStorage), this.helperInfo, this.settings.lang.volunteers, this.busy, (d: helperHistoryInfo, c) => c == d.courier);
+          }
+        },
+        {
+          name: 'הענק מתנה',
+          visible: () => this.settings.isSytemForMlt() && this.context.isAllowed(Roles.admin),
+          click: async () => {
+            let rows = this.helperInfo.selectedRows;
+            
+            if(rows.length == 0) {
+              this.dialog.Error('לא נבחרו מתנדבים');
+              return;
+            }
+
+            if (await this.context.openDialog(YesNoQuestionComponent, q => q.args = {
+              question: 'האם להעניק מתנה ל ' + rows.length + ' מתנדבים?'
+            }, q => q.yes)) {
+              if(await context.for(HelperGifts).count(g=>g.assignedToHelper.isEqualTo('')) >= rows.length) {
+                for await (const h of rows) {
```

Comment:
Hi, why "for await? I don't think that the await is needed at this point, since the selectedRows is a simple array.